src/main.rs

```diff
@@ -271,10 +299,19 @@ fn build_sysroot(
         eprintln!("A sysroot is now available in `{}`.", sysroot_dir.display());
     }
 
-    PathBuf::from(sysroot_dir)
+    sysroot_dir
 }
 
-fn cargo_careful(mut args: env::Args) {
+fn cargo_careful(args: env::Args) {
+    let mut args = args.peekable();
+
+    let use_sanitizer = if Some("--no-sanitizer") == args.peek().map(AsRef::as_ref) {
```

Comment:
I think we need to come up with some kind of concept for flags. I have an idea for that, but it needs a bit of design. I don't like mixing our flags with the cargo flag namespace and making the flag position-dependent is also rather ugly.

Reply:
I plan to make a concrete suggestion tomorrow that I will push to your branch.

Reply:
I have pushed something to master that hopefully makes it clear what I mean.